rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionImpl.java

```diff
@@ -57,6 +58,36 @@ public RxBleConnectionImpl(RxBleRadio rxBleRadio, RxBleGattCallback gattCallback
     }
 
     @Override
+    public Observable<Integer> requestMtu(int mtu) {
+        return privateRequestMtu(mtu, 20, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public Observable<Integer> requestMtu(int mtu, long timeout, TimeUnit timeUnit) {
+        return privateRequestMtu(mtu, timeout, timeUnit);
+    }
+
+    private Observable<Integer> privateRequestMtu(int mtu, long timeout, TimeUnit timeUnit) {
+        synchronized (discoveredServicesCache) {
+            final Observable<Integer> newObservable;
+            // performing actual discovery
+                newObservable = rxBleRadio
+                        .queue(new RxBleRadioOperationMtuRequest(
```

Comment:
I'm asking because if that's possible then with cache the method may return old values, which is no longer valid.